### PR TITLE
Higher level features

### LINKED ISSUE #102: Feature test cleanup
@StephHippo and I talked about this the other day. Our feature tests have a lot of repeated code in the step definitions. We should refactor our tests so they share more code.

Also, our features for course searching have an unnecessary background step..you don't need to authenticate to search for courses.

### PR DESCRIPTION
**Pull request by ajm188**
closes #102 

recommend stepping through one commit at a time

### CODE REVIEW

File: features/step_definitions/wishlist_steps.rb [hunk 1]
```diff
@@ -1,64 +1,45 @@
...
-  expect(course && WishlistItem.exists?(course: course, user: @current_user)).to be false
+When(/^I remove it from my wishlist$/) do
+  step 'I click the link "Remove from my wishlist"'
 end
 
-Then(/^I should have notifications turned on for ([A-Z]+) (\d+)$/) do |course_dept, course_number|
-  course = Course.where(department: course_dept, course_number: course_number)
-  expect(WishlistItem.where(course: course, user: @current_user).first.notify).to be true
+When(/^I turn (on|off) notifications$/) do |m|
+  step "I click the link \"Turn #{m} notifications\""
```
Comment (StephHippo) [inline features/step_definitions/wishlist_steps.rb]: Either remove all the "I click" steps or just use the original step names. Related to my previous comment.

**Comment by StephHippo** [comment]
Generally speaking, keep re-use at the helper-method/page level rather than in the steps. It reduces confusion later when people are trying to re-use steps. This is kind of a unfortunate part about feature tests since English is less explicit than code.

**Comment by ajm188** [comment]
@StephHippo updated. I agree this reads much better. Those steps weren't really capturing any useful abstractions.

File: features/wishlist.feature [hunk 2]
```diff
@@ -1,78 +1,51 @@
...
-    Given My wishlist is empty
-    And I have the course EECS 293 in my wishlist
-    And I have notifications turned on for EECS 293
-    When I visit "/wishlist"
-    Then I should see "Turn off notifications"
-    When I click the link "Turn off notifications"
-    Then I should have notifications turned off for EECS 293
-    And I should see "Turn on notifications"
-
-  Scenario: Toggle notifications in a wishlist with several courses
```
Comment (StephHippo) [inline features/wishlist.feature]: Help me out, did we drop a scenario?

Comment (ajm188) [inline features/wishlist.feature]: Yeah. It wasn't testing anything new.

**Comment by ajm188** [comment]
bump

**Comment by StephHippo** [comment]
:+1:

**Pull request merged**